quora-service/src/main/java/com/upgrad/quora/service/business/UserProfileBusinessService.java

```diff
@@ -0,0 +1,35 @@
+package com.upgrad.quora.service.business;
+
+import com.upgrad.quora.service.dao.UserDao;
+import com.upgrad.quora.service.entity.UserAuthTokenEntity;
+import com.upgrad.quora.service.entity.UserEntity;
+import com.upgrad.quora.service.exception.AuthorizationFailedException;
+import com.upgrad.quora.service.exception.UserNotFoundException;
+import java.time.ZonedDateTime;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class UserProfileBusinessService {
+
+  @Autowired
+  private UserDao userDao;
+
+  public UserEntity getUserProfile(final String userUuid, final String authorizationToken)
+      throws AuthorizationFailedException,
+      UserNotFoundException {
+
+    UserAuthTokenEntity userAuthTokenEntity = userDao.getUserAuthToken(authorizationToken);
+    UserEntity userEntity = userDao.getUser(userUuid);
+
+    if (userAuthTokenEntity.getLogoutAt() != null || userAuthTokenEntity.getExpiresAt()
```

Comment:
Please maintain the order of exceptions as per the requirement.

Reply:
Have re-worked the order of exception as per the requirement document in the latest commit. Please check.

Reply:
Ok.